bin/boilerplate.rs

```diff
@@ -93,6 +99,44 @@ impl Harness {
                             if !app.on_key(input) {
                                 running = false;
                             }
+                            info!("alt_pressed: {:?}", input);
+                            match input.virtual_keycode {
+                                Some(Key::LControl) => {
+                                    alt_pressed = input.state == ElementState::Pressed;
+                                }
+                                _ => {}
+                            }
+
+                        }
+                        glutin::WindowEvent::MouseWheel {delta, ..} => {
+                            app.on_mouse_wheel(delta)
+                        }
+                        glutin::WindowEvent::CursorMoved {position, ..} => {
+                            if mouse_pressed {
+                                match position {
+                                    (x, y) => {
+                                        if last_mouse_x >= 0.0 {
+                                            app.on_mouse_move(
+                                                x as f32 - last_mouse_x,
+                                                y as f32- last_mouse_y ,
+                                                alt_pressed,
+                                            );
+                                        }
+                                        last_mouse_x = x as f32;
+                                        last_mouse_y = y as f32;
+                                    }
+                                }
+                            }
+                        }
+                        glutin::WindowEvent::MouseInput {state, button, ..} => {
+                            if button == MouseButton::Left {
```

Comment:
this is quite specific and tailored for the current use in `level` binary. I think we should move some of that logic inside the `level` app, and keep the boiler plate generic, i.e. just pass down both `state` and `button` to the `Application` directly, let it deal with that.